backend/app/controllers/commentController.ts

```diff
@@ -23,6 +23,99 @@ class CommentController {
     else return res.status(404).json({ message: 'No comments found on that path' })
   }
 
+  public pinComment = async (req: Request | any, res: Response): Promise<Response> => {
+    const { id } = req.params
+
+    const post = await this.postDAO.getPostByPath(await this.dao.getPostPathByCommentId(id))
+    const profile = await this.profileDAO.getProfileByUsername(req.decoded.username)
+
+    if (profile.id === post.profile.id) {
+      await this.dao.pinComment(id)
+
+      return res.status(200).json({ message: 'Succesfully pinned comment' })
+    }
+
+    if (!post) {
+      return res.status(404).json({ message: 'Comment or post not found' })
+    }
+
+    return res.status(401).json({ message: 'Invalid user' })
+  }
+
+  public unpinComment = async (req: Request | any, res: Response): Promise<Response> => {
+    const { id } = req.params
+
+    const post = await this.postDAO.getPostByPath(await this.dao.getPostPathByCommentId(id))
+    const profile = await this.profileDAO.getProfileByUsername(req.decoded.username)
+
+    if (profile.id === post.profile.id) {
+      await this.dao.unpinComment(id)
+
+      return res.status(200).json({ message: 'Succesfully unpinned comment' })
+    }
+
+    if (!post) {
+      return res.status(404).json({ message: 'Comment or post not found' })
+    }
+
+    return res.status(401).json({ message: 'Invalid user' })
+  }
+
+  public getCommentLikes = async (req: Request, res: Response): Promise<Response> => {
+    const { id } = req.params
+
+    const foundCommentLikes = await this.dao.getCommentLikes(Number(id))
+
+    if (foundCommentLikes) return res.status(200).json(foundCommentLikes)
+    else return res.status(204).json([])
+  }
+
+  public getProfileLikedComment = async (req: Request | any, res: Response): Promise<Response> => {
+    const { id } = req.params
+
+    const profile = await this.profileDAO.getProfileByUsername(req.decoded.username)
+    const foundCommentLikes = await this.dao.getCommentLikes(Number(id))
+    let profileHasLikedComment = false
+
+    foundCommentLikes.forEach((commentLike: any) => {
+      if (commentLike.profile.id === profile.id) profileHasLikedComment = true
+    })
+
+    return res.status(200).json(profileHasLikedComment)
```

Comment:
Why not just do something like this?

```suggestion
    foundCommentLikes.forEach((commentLike: any) => {
      if (commentLike.profile.id === profile.id)
          return res.status(200).json(profileHasLikedComment)
    })

    return false
```

This way you don't loop an unnecessary amount of times